crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -1265,6 +1265,10 @@ impl Loader<HistoricalKey> for Db {
     async fn load(&self, keys: &[HistoricalKey]) -> Result<HashMap<HistoricalKey, Object>, Error> {
         use objects_version::dsl as v;
 
+        if keys.is_empty() {
```

Comment:
I think `HistoricalKey` and `PointLookupKey` are the only objects keys with this issue potentially since others use `eq_any` or individual db fetches. But I have not checked if this can happen to other types systematically.